Automattic/wp-calypso
# Jetpack Manage: Show some fields only on the sign up form - not on edit

Related to https://github.com/Automattic/jetpack-manage/issues/57

## Proposed Changes

* This PR passes a prop from the `<AgencySignupForm>` component through to the `<CompanyDetailsForm>` component that indicates the form is being used as a signup form, and then those form fields are visible (hidden otherwise). 
* The fields that are hidden are the "Company Type" and "Managed Sites", showing on the sign up form and hidden on `/`partner-portal/company-details` 
* The partner opt-in checkbox is hidden unless a new company type is selected, so remains hidden on this form.
* I needed to add a default value for state for `managed_sites` in `UpdateCompanyDetailsForm` as the form wouldn't save at all on the `/partner-portal/company-details/` page, unless selecting a value other than '1-5 'for managed sites, and then optionally selecting that 1-5 value again if that was the chosen option (it generated the following error on save: `The following details are required: managed sites` ). Because of that, hiding the field still resulted in the same error until fixed.

## Testing Instructions

* With this branch checked out, open the agency sign up form in development mode - `http://jetpack.cloud.localhost:3000/agency/signup` - with a new WP.com account.
* The following fields should be visible: Company Type ('Which answer below best describes your company:'), Partner Opt-in checkbox under the options to best describe your company (if you've selected one of the top two company types), Manage Sites ('How many sites do you manage?')
* Open the 'Company Edit' page for an account that already is listed as a partner: `http://jetpack.cloud.localhost:3000/partner-portal/company-details`
* The above mentioned fields should all be hidden.
* Make an edit to any of the existing fields, and then either check the network tab in the developer inspector to see the payload or visit the tracks live view (or use Tracks Vigilante) and watch the `calypso_partner_portal_create_partner_submit` event for your submission, and check the props. There should be no referrer prop passed through.
